### PR TITLE
Use katdal.open(preselect=...) for better performance/memory

### PR DESCRIPTION
**Pull request by bmerry**

### CODE REVIEW

File: katsdpimager/loader.py [hunk 1]
```diff
@@ -10,10 +10,26 @@ def _register_loader(loader_class):
...
+    Parameters
+    ----------
+    filename : str
+        Name of the data set to load (does not necessarily have to be a filename
+        e.g., it could be an URL).
+    options : Dict[str, Any]
+        Loader-specific command-line options.
+    start_channel,stop_channel : int, optional
+        Channel range of interest. This serves as a hint only, to optimise
+        loading, and does not change the channel numbers used to interact with
```
Comment (ludwigschwardt) [inline katsdpimager/loader.py:27]: That would have been an interesting option for `preselect` itself too.

File: katsdpimager/loader_katdal.py [hunk 2]
```diff
@@ -131,6 +135,10 @@ def __init__(self, filename, options):
             raise ValueError('Subarray {} is out of range'.format(args.subarray))
         if args.spw < 0 or args.spw >= len(self._file.spectral_windows):
             raise ValueError('Spectral window {} is out of range'.format(args.spw))
+        # Frequency-dependent private attributes contain data only for
+        # start_channel:stop_channel. The public interface adjusts by
+        # _start_channel to present channels 0:stop_channel.
```
Comment (ludwigschwardt) [inline katsdpimager/loader_katdal.py:140]: Will the loader ever be asked for e.g. channel 0? Or is that prevented by `args.start_channel`? Still trying to wrap my head around the difference in channel count...

**Review by ludwigschwardt — APPROVED** [review]

File: katsdpimager/loader_katdal.py (hunk 2 above)
Comment (bmerry) [reply katsdpimager/loader_katdal.py:140]: The loader will only ask about channels in start_channel:stop_channel. The constructor arguments are a hint about the channels that the loader will be asked about in future.

Comment (ludwigschwardt) [reply katsdpimager/loader_katdal.py:140]: Right, that's what I suspected.

**Pull request merged**